### PR TITLE
Census address format

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
The address format is changing for census attributes, including an address line 3. This PR brings the [actionsvc-api](https://github.com/ONSdigital/rm-actionsvc-api) classes into this project where they were previously imported as a dependency. This allows us to change them without worrying about setting up a separate census version building to an artifactory (though we will have to make sure it remains identical accross the other projects which use then transport classes). Also renames some social specific code to census where it has now been repurposed.

# What has changed
- Bring in actionsvc-api transport classes, xsd's and tests
- Add pom goals to generate sources from xsd's
- Add address line 3 to the action address xsd
- Add test for sample attributes request decorator
- Rename repurposed social specific code to census and update tests accordingly 

# How to test?
This is waiting on changes to the casesvc, actionexportersvc and sample loader before it can easily be manually tested. For now you can run the integration tests and make sure they make sense.

# Links
https://trello.com/c/e9iWXgje/517-rmc-83-load-hh-sample-and-generate-print-file-to-census-spec-13

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[RMC-83 Load HH sample and generate print file to Census spec [13]](https://trello.com/c/e9iWXgje/517-rmc-83-load-hh-sample-and-generate-print-file-to-census-spec-13)

**Comment by philwhiles** [comment]
Tuppence from the original developer ... the api projects were there for good reason - any other service wanting to call the action service will need its own copy of all those DTOs or its own copy of the XSD. Copy and paste programming **never** a good idea, because we know where that will lead. And the Case service for instance cannot have a dependency on action-service without pulling **all** of its code into its own jar.

**Comment by davidmort** [comment]
whilst i'd agree that copy and paste is not always a good idea, i'd also argue that shared models between microservices is a bad idea, we much prefer a bit of duplication over tight coupling.

the reality of the api projects is that the code surrounding them still has to change when the apis change, and also the api's don't change enough to warrant having a specific pipeline/deployment for them or the extra overhead of managing the dependency. it's far easier just to wrap the code up in the app.  

Our intention is to decouple the RM services, reduce the point to point apis calls and replace with an event driven approach hence reducing the need for sharing api models even further

File: src/main/java/uk/gov/ons/ctp/response/action/representation/ActionType.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package uk.gov.ons.ctp.response.action.representation;
+
+public enum ActionType {
+  BSNOT("BSNOT"),
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/ctp/response/action/representation/ActionType.java:4]: Can the old action types be removed?

**Review by LukeLoze1 — CHANGES_REQUESTED** [review]
Looks ok to me.  Works with my local system.  Only issue would be to remove obviously unused actionTypes

File: src/main/java/uk/gov/ons/ctp/response/action/representation/ActionType.java (hunk 1 above)
Comment (AdamHawtin) [reply src/main/java/uk/gov/ons/ctp/response/action/representation/ActionType.java:4]: Not without deleting or re-writing a number of tests, I've created a ticket for this.

**Review by liamtoozer — APPROVED** [review]
Looks good to me. Ran tests locally from the other branches as well.

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java [hunk 2]
```diff
@@ -148,17 +145,9 @@ private ActionProcessingService getActionProcessingService(Action action) {
...
-        return socialActionProcessingService;
-
-      case B:
-      case BI:
-        return businessActionProcessingService;
-
-      default:
-        throw new UnsupportedOperationException("Sample Type: " + caseType + " is not supported!");
+    if (caseType == SampleUnitDTO.SampleUnitType.H) {
+      return socialActionProcessingService;
```
Comment (liamtoozer) [inline src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java]: Is it worth changing this name to `censusActionProcessingService`?

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

**Review by LukeLoze1 — APPROVED** [review]

**Review by hbrace — APPROVED** [review]

**Pull request merged**